lib/guess_validator.rb

```diff
@@ -0,0 +1,62 @@
+class GuessValidator
+  def initialize
+    @colors = ["R", "O", "Y", "G", "B", "P"]
+    @validated = false
+  end
+
+  def validate(guess)
+    @validated = true
+    guess = guess.to_s
+    errors = []
+
+    if numbers?(guess)
+      errors << "Numbers are not allowed."
+    end
+
+    if punctuation?(guess)
+      errors << "Punctuation is not allowed."
+    end
+
+    if too_short?(guess)
+      errors << "Your guess is too short. Please enter 4 letters."
+    end
+
+    if too_long?(guess)
+      errors << "Your guess is too long. Only 4 letters are allowed."
+    end
+
+    unless (numbers?(guess) || punctuation?(guess))
+      if invalid_colors?(guess)
+        errors << "Your guess has invalid colors. Please use only available colors."
+      end
+    end
+
+    errors
+  end
+
```

Comment:
Great work here. We can further clean up this method by encapsulating our `if` statements. You can create a method like `run_validations` and run all your `if` statements in that while just making one call to `run_validations` in your `validate` method. It'll make your method a lot easier to read and understand I think.